### ISSUE TITLE
Dynamically changing img based on "text"

### ISSUE THREAD

**Issue by vikasprogrammer** [joshvillbrandt/ng-holder#2]
Is it possible to dynamically change `img` based on the text set by controller?

**Comment by joshvillbrandt** [comment]
Not the way it is currently written. But something like this could work assuming you are using `data-src`:

``` javascript
module.directive('holder', [
  function() {
    return {
      link: function(scope, element, attrs) {
        if(attrs.holder) {
          attrs.$set('data-src', attrs.holder);
        }
        scope.$watch('data-src') function() {
          Holder.run({images:element[0]});
        });
      }
    };
  }]);
```

I kinda like how the current implementation doesn't produce a watcher though. I'd like to find a way to have both before committing this to the repo. Thoughts?

**Comment by vikasprogrammer** [comment]
You are right, having a watcher depends upon the implementation and may not be suitable for every use case. Can we provide a configuration variable or a directive to enable watching, like :

`<img holder data-src="" holder-watch>`

**Comment by atais** [comment]
IMO the watch is actually required.

I am now facing the issue, that the holder can not read dynamic `$scope` variables, for example:

```javascript
<img holder
     data-src="holder.js/{{width}}x{{height}}?text={{width}}x{{height}}.">
```
even though in DOM inspector the variables are set, the holder must have initialized earlier and is not visible. By adding the watch section it works great.
